src/main/java/com/vmware/vipclient/i18n/messages/service/ComponentService.java

```diff
@@ -70,37 +69,87 @@ private void refreshCacheItem(final MessageCacheItem cacheItem, Iterator<DataSou
 		}
 	}
 
+	@Deprecated
+	public MessageCacheItem getMessagesOld() {
```

Comment:
@deprecated is there to mark that this method shall also be removed once we remove the other deprecated APIs (TranslationMessage.getStrings, getString, getString2). This is also marked as deprecated so that it won't be used by any other developer for future enhancements/features. 

The name is changed only because I wanted to use "getMessages" for the new method as it is simple and descriptive.

Reply:
A deprecated interface is an interface existing before and will be deprecated. But this method is a new interface/method, so this is a wrong use of deprecated tag.
You should change the name back or remove the tag, right?

Reply:
The compatibility has not been broken. If you trace the code from getMessagesOld, it follows the same logic as before. No change. 

getMessagesOld calls getMessages(fallbackLocalesIter), which calls getMessages(fallbackLocalesIter, true)
Inside getMessages fallbackLocalesIter, true), it will enter either one of the following:
1. if (cacheService.isContainComponent()) in line 158 if already cached
2. if (waitForFetch) in line 173 if not yet in cache. It will wait for the fetch to come back. It won't use a fallback.
This was the behavior before this PR, and we are keeping it for getMessagesOld. No change.

If still unclear, let's discuss offline soon.

Reply:
getMessagesOld is a new interface. Don't include it into compatibility.

Reply:
@Xiaochao8 ,did you read this: As I said above, please see https://confluence.eng.vmware.com/display/GQ/Client+Library+Workflow+-+Backwards+Compatibility+Strategy